llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/LlapLoadGeneratorService.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.llap.daemon.impl;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.service.AbstractService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Random;
+
+/**
+ * Extra load generator service for LLAP.
+ */
+public class LlapLoadGeneratorService extends AbstractService {
+  private static final Logger LOG = LoggerFactory.getLogger(LlapLoadGeneratorService.class);
+  private static final long INTERVAL = 10;
+  private float threshold;
+  private String[] hostNames;
+  private int processors;
+  private Thread[] threads;
+
+  public LlapLoadGeneratorService() {
+    super("LlapLoadGeneratorService");
+  }
+
+  @Override
+  protected void serviceInit(Configuration conf) throws Exception {
+    super.serviceInit(conf);
+    threshold = HiveConf.getFloatVar(conf, HiveConf.ConfVars.HIVE_TEST_LOAD_UTILIZATION);
+    hostNames = HiveConf.getTrimmedStringsVar(conf, HiveConf.ConfVars.HIVE_TEST_LOAD_HOSTNAME);
+    LOG.info("LlapLoadGeneratorService init with {} {}", threshold, hostNames);
+  }
+
+  protected void serviceStart() throws UnknownHostException {
```

Comment:
@Override added

Exception - it is very unlikely that we get an exception here: "if the local host name could not be resolved into an address.". We specifically wanted to configure load for the cluster and we are unable to identify one of the nodes. I think it might be reasonable to fail to start the Daemon in this case? If I am running the tests I would like to face this issue ASAP, and not after the my 10 hour testsuite is finished and checking the results :)

Still there can be good reasons to try to start it anyway. What do you recommend? Shall we log a warn message only and continue? What do we usually do in case of LlapDaemons?